dask/dataframe/shuffle.py

```diff
@@ -545,15 +550,17 @@ def collect(p, part, meta, barrier_token):
 
 
 def set_partitions_pre(s, divisions):
-    partitions = pd.Series(divisions).searchsorted(s, side="right") - 1
-    partitions[(s >= divisions[-1]).values] = len(divisions) - 2
+    partitions = divisions.searchsorted(s, side="right") - 1
+    if hasattr(partitions, "index"):
```

Comment:
Does `is_series_like(partitions)` work here rather than the hasattr? That may be more robust.

Reply:
Pandas also has a (private) `extract_array` method that gets an array from a Series or Index (and returns other stuff though untouched). We can make that public if it's generally useful.

Reply:
Your `is_series_like(partitions)` suggestion does the trick here - Thanks!